### PR TITLE
Add Match and Boardgame pages

### PR DESCRIPTION
**Pull request by peoplenarthax**
This still has a lot of work in progress but I though it was a nice idea to merge it down to can continue working in a cleaner direction.

This PR adds:
- Prototyping components (UglyCards, UglyGrid, UglyInputs...)
- Schema Validator Generator for boardgames
- Schema validation reader and add match form generation
...

Future work:
- Move from prototyping components to real reausable components for layout together with the theme framework
- Clean page structure to make it easy to compose
- Add isWinner as a default player result field and make it easy to mark in the add match form
- Experiment for desktop layout
- Create User journeys for user actions

### CODE REVIEW

File: .prettierrc [hunk 1]
```diff
@@ -2,5 +2,6 @@
   "semi": false,
   "singleQuote": true,
   "trailingComma": "all",
-  "bracketSpacing": true
+  "bracketSpacing": true,
+  "parser": "typescript"
```
Comment (beeequeue) [inline .prettierrc:6]: this shouldnt be necessary?

File: packages/ui/package.json [hunk 2]
```diff
@@ -89,7 +91,7 @@
   },
   "lint-staged": {
     "*.{js,json,yml}": [
-      "prettier --write",
+      "prettier --write --parser typescript",
```
Comment (beeequeue) [inline packages/ui/package.json]: same here, this should make formatting the yml files fail

File: packages/ui/src/pages/boardgame/components/property-form.tsx [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+      setProperty({ name: e.target.value, type: property.type })
+    },
+    [setProperty, property],
+  )
+
+  return (
+    <Container>
+      <FieldContainer>
+        <Input
+          placeholder="Property Name"
```
Comment (beeequeue) [inline packages/ui/src/pages/boardgame/components/property-form.tsx:113]:
```suggestion
          placeholder={name}
```
?

File: packages/ui/src/pages/boardgame/generateSchemaValidation.ts [hunk 4]
```diff
@@ -0,0 +1,54 @@
+import { Property } from '@/pages/boardgame/components/property-form'
```
Comment (beeequeue) [inline packages/ui/src/pages/boardgame/generateSchemaValidation.ts]: this file's name is camelCase

File: packages/ui/src/pages/boardgame/generateSchemaValidation.ts [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+        pattern:
+          '^[0-9a-fA-F]{8}\\-[0-9a-fA-F]{4}\\-[0-9a-fA-F]{4}\\-[0-9a-fA-F]{4}\\-[0-9a-fA-F]{12}$',
+      },
+    },
+  )
+  const propertyArray = Object.keys(formattedProperties)
+    .map(name => `"${name}"`)
+    .toString()
+
+  return `
```
Comment (beeequeue) [inline packages/ui/src/pages/boardgame/generateSchemaValidation.ts]: Could this be a stringified object instead of a string

**Review by beeequeue — CHANGES_REQUESTED** [review]
Only thing I really want is the new pages added to the routes :P

File: packages/ui/src/pages/boardgame/components/property-form.tsx (hunk 3 above)
Comment (peoplenarthax) [reply packages/ui/src/pages/boardgame/components/property-form.tsx:113]: at this point property names passed down have the shape "property-0" ... Property name is a better placeholder

File: packages/ui/src/pages/boardgame/generateSchemaValidation.ts (hunk 5 above)
Comment (peoplenarthax) [reply packages/ui/src/pages/boardgame/generateSchemaValidation.ts]: that is actually better, and easier to test :D!

File: packages/server/src/types/json-schema.ts [hunk 6]
```diff
@@ -41,19 +41,21 @@ export interface JsonSchemaNumber extends JsonSchemaPropertyBase {
...
-    pattern: string
-  }
-  minProperties?: number
-  maxProperties?: number
-  properties?: {
-    [key: string]: JsonSchemaProperty
-  }
-  dependencies?: {
-    [key: string]: string[]
+  schema: {
```
Comment (beeequeue) [inline packages/server/src/types/json-schema.ts]: This isn't correct. This is the type for how to write a JSON object property using JSON Schema.

File: packages/ui/package.json (hunk 2 above)
Comment (beeequeue) [reply packages/ui/package.json]: this is still here

File: packages/server/src/graphql/snapshot.graphql [hunk 7]
```diff
@@ -73,9 +73,11 @@ type Mutation {
 
 type Query {
   boardgame(uuid: ID!): Boardgame
+  boardgames: [Boardgame!]
   club(uuid: ID!): Club
   match(uuid: ID!): Match
   user(uuid: ID!): User
+  users: [User!]
```
Comment (beeequeue) [inline packages/server/src/graphql/snapshot.graphql]: theyre gonna be rebased, but they are currently optional

File: packages/server/src/types/json-schema.ts [hunk 8]
```diff
@@ -41,20 +41,20 @@ export interface JsonSchemaNumber extends JsonSchemaPropertyBase {
...
-  }
-  minProperties?: number
-  maxProperties?: number
-  properties?: {
-    [key: string]: JsonSchemaProperty
-  }
-  dependencies?: {
-    [key: string]: string[]
-  }
+    type: 'object'
```
Comment (beeequeue) [inline packages/server/src/types/json-schema.ts]: prettier didnt run

File: packages/ui/src/utils/game-schema-types.ts [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+const toPropertyType = (value: string): PropertyType => {
+  const propertyTypes = [PropertyType.STRING, PropertyType.NUMBER]
+
+  return (
+    propertyTypes.find(prop => prop.toLowerCase() === value) ||
+    PropertyType.STRING
+  )
+}
+
+type SCHEMA = {
```
Comment (beeequeue) [inline packages/ui/src/utils/game-schema-types.ts]: CapitalCase

File: packages/ui/package.json [hunk 10]
```diff
@@ -108,7 +108,7 @@
     "lint": "eslint \"src/**/*.{ts,tsx}\"",
     "storybook": "start-storybook -p 6006",
     "build-storybook": "build-storybook -o .storybook/dist",
-    "test": "jest src",
+    "test": "NODE_ENV=test jest ui -i",
```
Comment (beeequeue) [inline packages/ui/package.json]: Is `-i` needed? Doesn't it just make tests run serially instead of in parallel?

Comment (peoplenarthax) [reply packages/ui/package.json]: true, tried the same command as in the server

File: packages/ui/package.json [hunk 11]
```diff
@@ -106,7 +108,7 @@
     "lint": "eslint \"src/**/*.{ts,tsx}\"",
     "storybook": "start-storybook -p 6006",
     "build-storybook": "build-storybook -o .storybook/dist",
-    "test": "jest ui",
+    "test": "NODE_ENV=test jest ui -i",
```
Comment (peoplenarthax) [inline packages/ui/package.json]:
```suggestion
    "test": "NODE_ENV=test jest ui",
```

**Review by beeequeue — APPROVED** [review]

**Pull request merged**